hibernate-core/src/main/java/org/hibernate/internal/AbstractSharedSessionContract.java

```diff
@@ -392,13 +392,14 @@ public Transaction getTransaction() throws HibernateException {
 
 	@Override
 	public Transaction accessTransaction() {
-		if ( this.currentHibernateTransaction == null || this.currentHibernateTransaction.getStatus() != TransactionStatus.ACTIVE ) {
+		if ( this.currentHibernateTransaction == null
```

Comment:
I'm not sure this is correct from a JPA perspective.  JPA says that `EntityTransaction` is used to "serially ... begin and commit multiple transactions".  It sure seems like JPA expects that the same `EntityTransaction` be returned each time that `EntityManager#getTransaction` is called:

    EntityTransaction t = manager().getTransaction();
    assertSame( t, manager().getTransaction() );
    assertFalse( t.isActive() );
    t.begin();
    assertSame( t, manager().getTransaction() );
    assertTrue( t.isActive() );
    t.commit();
    assertSame( t, manager().getTransaction() );
    assertFalse( t.isActive() );
    manager().close();
    assertSame( t, manager().getTransaction() );
    assertFalse( t.isActive() );

As it stands we create a new `EntityTransaction` as soon the transaction is accessed after the commit.

I'm not sure I can say that the spec requires this (^^) behavior though, and the TCK surely seems to not enforce it.